src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -183,6 +185,193 @@ def list_vaults(client, resource_group_name=None):
+
+    curr_identity_details = vault_details.identity
+    curr_identity_type = 'none'
+
+    if curr_identity_details is not None:
+        curr_identity_type = curr_identity_details.type.lower()
+    user_assigned_identity = None
+
+    if identity_type is not None:
+        if remove_user_assigned or remove_system_assigned:
+            raise MutuallyExclusiveArgumentError("Addition and Deletion of identities are not possible at same time.")
+
+        identity_type = identity_type.replace(" ", "").lower()
+
+        if identity_type in ["none", "systemassigned"]:
+            if identity_id is not None:
+                raise ArgumentUsageError("--identiy-id paramter is only supported for UserAssigned identities.")
+
+            if curr_identity_type in ["systemassigned, userassigned", "userassigned"]:
+                if identity_type == "systemassigned":
+                    identity_type = 'systemassigned,userassigned'
+        elif identity_type == 'userassigned':
+            if identity_id is None:
+                raise RequiredArgumentMissingError("Please provide identity id using --identity-id parameter.")
+
+            userid = UserIdentity()
+            user_assigned_identity = dict()
+            for element in identity_id:
+                user_assigned_identity[element] = userid
+            if curr_identity_type in ["systemassigned", "systemassigned, userassigned"]:
+                identity_type = 'systemassigned,userassigned'
+
+    elif remove_system_assigned or remove_user_assigned:
+        return remove_identities(client, resource_group_name, vault_name, curr_identity_details,
+                                 curr_identity_type, identity_id, remove_user_assigned,
+                                 remove_system_assigned)
+    else:
+        raise CLIError(
+            """
+            Invalid parameters, no operation specified.
+            """)
+
+    identity_data = IdentityData(type=identity_type, user_assigned_identities=user_assigned_identity)
+    vault = PatchVault(identity=identity_data)
+    return client.begin_update(resource_group_name, vault_name, vault)
+
+
+def remove_identities(client, resource_group_name, vault_name, curr_identity_details, curr_identity_type,
+                      identity_id, remove_user_assigned, remove_system_assigned):
+    identity_type = None
+    user_assigned_identity = None
+    if remove_user_assigned and remove_system_assigned:
+        raise MutuallyExclusiveArgumentError(
+            """
+            Both system and user assigned identities can't be removed at same time.
+            """)
+
+    if remove_system_assigned:
+        if identity_id is not None:
+            raise MutuallyExclusiveArgumentError(
+                """
+                --identiy-id paramter is only supported for UserAssigned identities
+                """)
+        if curr_identity_type not in ["systemassigned", "systemassigned, userassigned"]:
+            raise CLIError(
```

Comment:
Please use a specific error class. Look for the error class which suits best in this scenario from azclierror.py